### ISSUE TITLE
Prevent users from adding incorrect modifiers to arrays

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#11498]
With PR #11019 it is now possible to set each and final for array modifiers. However, it is still possible to input an incorrect modifier, which will then generate errors at compile time. It is also unclear from the dialog whether the modifier actually applies to an array or scalar, which makes it difficult to avoid errors (though default values can help).

We should
- [x] add the dimension of the parameter to its name in the parameter dialog, e.g. `w0[3]` instead of `w0` for the initial value of the angular speed in multibody systems
- [ ] refuse to accept a scalar value for arrays, with and error message popping up such as "This is an array parameter. Either provide an array of values of the same dimension or set 'each' with the context menu if you want to provide a single scalar value to all the elements of the array"

**Comment by casella** [comment]
Regarding the second item, I'm not sure if OMEdit should try to figure out itself, or if we should just tentatively pass the modeified model to the inst-based interface and reject it if there is some error. The only problem with the latter approach is that if the model has _other_ issues besides the wrong applied modifiere, the feedback could be confusing.

@adeas31 what do you think?

**Comment by adeas31** [comment]
I think the current behavior kind of handles the second item. @perost improved this in 0e8be8760a3ff89d2cfa8f752a60bae42da1acf1 when I added support for `each` in the GUI. Now the instance-based interface doesn't fail and gives a warning about missing each. So its same as you mentioned in the comment, if there are other errors then they will also be shown. The only difference is that we don't restrict instead we allow and then give a warning.

**Comment by casella** [comment]
OK. What I mean is, if the inst-based interface detects a wrong modifier, OMEdit should refuse to apply it to the source code outright, so that it is basically impossible to generate wrong code with OMEdit regarding this point. Will this be the case?

**Comment by adeas31** [comment]
No, this is not the case. I will discuss with @perost if we can prevent this somehow.

**Comment by adeas31** [comment]
> add the dimension of the parameter to its name in the parameter dialog, e.g. w0[3] instead of w0 for the initial value of the angular speed in multibody systems

Do you want to see the typed dimension or the expression as defined in the model?

**Comment by casella** [comment]
I guess I'd go for the typed dimension. I mean, I'd prefer to see `w0[3]` rather than `w0[N]` because it may not be immediately apparent what the value of `N` is. I hope I understood your question correctly 😅

**Comment by casella** [comment]
First task completed in 1.22.1, see #11710. Second part postponed to 1.23.0.

**Comment by adeas31** [comment]
@casella I believe the second part of this issue is not needed since we had fixed #13636. What do you think?

**Comment by casella** [comment]
Of course, that request is now obsolete. We can close this for 1.22.1.
